native/packages/react-native-bpk-component-badge/stories.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { storiesOf } from '@storybook/react-native';
+import { View, StyleSheet } from 'react-native';
+import { colorGray900, spacingSm } from 'bpk-tokens/tokens/base.react.native';
+
+import BpkBadge from './index';
+import { BADGE_TYPES, BADGE_DOCKED_TYPES } from './src/BpkBadge';
+import { StorySubheading } from '../../storybook/TextStyles';
+
+const style = StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    justifyContent: 'flex-start',
+  },
+  badge: {
+    margin: spacingSm,
+  },
+  dark: {
+    backgroundColor: colorGray900,
+  },
+});
+function generateBadgeStory(docks, isDark = false) {
+  const types = isDark
+    ? [BADGE_TYPES.light, BADGE_TYPES.inverse, BADGE_TYPES.outline]
+    : [BADGE_TYPES.success, BADGE_TYPES.warning, BADGE_TYPES.destructive];
+  const containerStyle = isDark
+    ? [style.container, style.dark]
+    : style.container;
+  const badges = types.map(i => (
+    <BpkBadge
+      key={i}
+      style={style.badge}
+      message="Badge"
+      docked={docks}
+      type={i}
+    />
+  ));
+  return <View style={containerStyle}>{badges}</View>;
+}
+
+storiesOf('BpkBadge', module)
+  .add('docs:default', () => (
+    <View>
+      <StorySubheading>Default</StorySubheading>
+      {generateBadgeStory(null)}
```

Comment:
This doesn't look great - i know it's just for the story's but it'd be much easier to understand if it looked like:
```
generateBadgeStory({
  dark: false,
})

generateBadgeStory({
  docked: 'left',
  dark: true,
})
```

Reply:
fair enough, easy to change